src/bolt/utils.ts

```diff
@@ -59,55 +63,123 @@ export async function getReactionData (event: ReactionAddedEvent | ReactionRemov
 
     const getMessageInfo = (data: ConversationsRepliesResponse): ReactionData => {
         const message = (data.messages ?? [])[0] as GenericMessageEvent;
+        let pullRequestLink = '';
+
+        // Try to parse github pull request from text message if available.
+        if (message.text) {
+            const match = /https:\/\/github.com\/[^ ]+?\/pull\/\d+/.exec(message.text);
+            pullRequestLink = match ? match[0] : '';
+        }
+
+        // If there is no PR info, try to check the message attachments, message is coming from github event.
+        if (!pullRequestLink.length && message.attachments?.length) {
+            const {title} = message.attachments[0];
+            if (title) {
+                const match = /https:\/\/github.com\/[^ ]+?\/pull\/\d+/.exec(title);
+                pullRequestLink = match ? match[0] : '';
+            }
+        }
+
         return {
             slackMsgText: message.text,
             slackMsgId: message.client_msg_id,
             slackMsgUserId: message.user,
-            slackThreadTs: message.thread_ts ?? message.ts
+            slackThreadTs: message.thread_ts ?? message.ts,
+            pullRequestLink,
         } as ReactionData;
     };
     const botUserId = await getBotUserId();
-    let messageInfo = null;
+    let messageInfo = getMessageInfo(result);
+
+    if (!result.client_msg_id && botUserId === event.item_user && !messageInfo.pullRequestLink.length &&
```

Comment:
I propose to rewrite a bit this part: 
By possibility group this checks. 
If it's not possible create util method isPresent(object, fieldName) and  isEmpty(object, fieldName).
function isPresent  (object, fieldName): boolean  {
return (object.fieldName) ? true: false;
}
function isEmpty (object, fieldName)  {
return object?.fieldName && object.fieldName != null &&  object.fieldName !="" object.fieldName.length > 0

So it will look like.
const messageIdEmpty = isEmpty(result, client_msg_id )
const pullRequestLinkEmpty = isEmpty(messageInfo, pullRequestLink)
const botUserIDEqualUser =  botUserId === event.item_user
const slackMessageTsNotEqualThreadTs =  messageInfo.slackMsgTs !== messageInfo.slackThreadTs
 if (
 !messageIdEmpty  
 && !pullRequestLinkEmpty
  && botUserIDEqualUser 
   && slackMessageTsNotEqualThreadTs 
  )